src/ConnectionAutomaton.cpp

```diff
@@ -87,6 +86,7 @@ void ConnectionAutomaton::connect(
       keepaliveTimer_->start(shared_from_this());
     }
   }
+  onConnected_();
```

Comment:
by changing this order, we might get into the situation that sending frames on line 82 may terminate socket and the client code would get onClose before onConnected.